test/functional/testlib.bash

```diff
@@ -60,6 +60,81 @@ generate_random_name() {
 
 }
 
+# Creates public and private key
+# Parameters:
+# - key_path: path to private key
+# - cert_path: path to public key
+generate_certificate() {
+
+	local key_path=$1
+	local cert_path=$2
+
+	# If incorrect number of parameters are received show usage
+	if [ $# -ne 2 ]; then
+		cat <<-EOM
+			Usage:
+			    generate_certificate <key_path> <cert_path>
+			EOM
+		return
+	fi
+
+	# generate self-signed public and private key
+	openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 \
+		-keyout "$key_path" -out "$cert_path" \
+		-subj "/C=US/ST=Oregon/L=Portland/O=Company Name/OU=Org/CN=localhost"
+
+}
+
+# Creates the trusted certificate store and adds one certificate
+# Parameters:
+# - cert_path: path of certificate to add to trust store
+create_trusted_cacert() {
+
+	local cert_path=$1
+	local counter=0
+	local subj_hash
+
+	# If incorrect number of parameters are received show usage
+	if [ $# -ne 1 ]; then
+		cat <<-EOM
+			Usage:
+			    create_trusted_cacert <certificate_path>
+			EOM
+		return
+	fi
+
+	# only one test can use the trusted certificate store at the same time to
+	# avoid certificate contamination from other tests and race conditions.
+	until mkdir "$CACERT_DIR"; do
+		sleep 1
+		if [ "$counter" -eq "1000" ]; then
+			echo "Timeout: creating trusted certificate"
+			return 1
+		fi
+		counter=$[$counter + 1]
```

Comment:
I hadn't seen a variable being increased like that in bash before, I just learned something new.
I usually see it like this: counter=$((counter+1))